### PR TITLE
Adds additional queued statuses to avoid requeueing.

### LINKED ISSUE #390: Deposit race condition
Here's what I think is happening:
1. At time _t_, cron job calls `gb:deposit` which queues books to be deposited. Let's say one of those deposit jobs is job _a_ for book _1_.
2. At time _t_+30, cron job calls `gb:deposit` which queues books to be deposited. Job _a_ still hasn't been performed yet, so status of book _1_ is still `DOWNLOAD_SUCCEEDED` and job _b_ is queued for book _1_.
3. Job _a_ is performed and status of book _1_ is changed to `DEPOSIT_SUCCEEDED`.
4. Job _b_ is performed, but since status of book _1_ is not `DOWNLOAD_SUCCEEDED`, raises `RuntimeError: Only books that have status DOWNLOAD_SUCCEEDED can be deposited. 20501136794 has status DEPOSIT_SUCCEEDED`. (For example: https://app.honeybadger.io/projects/67990/faults/61027618).

### PR DESCRIPTION
**Pull request by justinlittman**
closes #390

## Why was this change made?
To avoid requeueing downloads or deposits that are already queued.

## Was the documentation (README, DevOpsDocs, wiki) updated?
Yes.

## Does this change affect how this application integrates with other services?
If so, please confirm change was tested on stage and/or test added to sul-dlss/infrastructure-integration-test.
No.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,13 +134,15 @@ rake gb:stats                            # Print out book statistics
...
    * If downloading failed, the status for the book record is set to `DOWNLOAD_FAILED` and the reason is recorded.
 6. Deposit (gb:deposit): For each book that is `DOWNLOAD_SUCCEEDED`, deposit into SDR via SDR Deposit API.
+   * When the deposit is queued, the status for the book record set set to `DEPOSIT_QUEUED`.
    * If deposit succeeded, the status for the book record is set to `DEPOSIT_SUCCEEDED` and the druid is recorded.
    * If deposit failed, the status for the book record is set to `DEPOSIT_FAILED` and the reason is recorded.
 
-Thus, the progression of book statuses is `READY` > `FETCH_MARCXML_SUCCEEDED` (or `FETCH_MARCXML_FAILED`) > `CONVERSION_REQ_SUCCEEDED` (or `CONVERSION_REQ_FAILED`) > `DOWNLOAD_READY` > `DOWNLOAD_SUCCEEDED` (or `DOWNLOAD_FAILED`) > `DEPOSIT_SUCCEEDED` (or `DEPOSIT_FAILED`).
+Thus, the progression of book statuses is `READY` > `FETCH_MARCXML_SUCCEEDED` (or `FETCH_MARCXML_FAILED`) > `CONVERSION_REQ_SUCCEEDED` (or `CONVERSION_REQ_FAILED`) > `DOWNLOAD_READY` > `DOWNLOAD_QUEUED` > DOWNLOAD_SUCCEEDED` (or `DOWNLOAD_FAILED`) > `DEPOSIT_QUEUED` >`DEPOSIT_SUCCEEDED` (or `DEPOSIT_FAILED`).
```
Comment (mjgiarlo) [inline README.md]: Is there a backtick missing before DOWNLOAD_SUCCEEDED?

Comment (justinlittman) [reply README.md]: Fixed.

File: app/services/fetcher_service.rb [hunk 2]
```diff
@@ -127,8 +129,9 @@ def download
   # @raise [FetcherServiceError] if download or verification failed
   # rubocop:disable Metrics/MethodLength
   def download_single(book)
-    unless book.status == Book::DOWNLOAD_READY
-      raise "Only books that have status DOWNLOAD_READY can be downloaded. #{book.barcode} has status #{book.status}"
+    unless [Book::DOWNLOAD_READY, Book::DOWNLOAD_QUEUED].include?(book.status)
```
Comment (mjgiarlo) [inline app/services/fetcher_service.rb]: Should the comment on line 126 be updated to reflect this change?

File: app/services/fetcher_service.rb [hunk 3]
```diff
@@ -173,8 +178,9 @@ def deposit
   # @raise [FetcherServiceError] if deposit failed
   # rubocop:disable Metrics/MethodLength
   def deposit_single(book)
-    unless book.status == Book::DOWNLOAD_SUCCEEDED
-      raise "Only books that have status DOWNLOAD_SUCCEEDED can be deposited. #{book.barcode} has status #{book.status}"
+    unless [Book::DOWNLOAD_SUCCEEDED, Book::DEPOSIT_QUEUED].include?(book.status)
```
Comment (mjgiarlo) [inline app/services/fetcher_service.rb]: Change comment on line 175?

File: app/services/fetcher_service.rb (hunk 2 above)
Comment (justinlittman) [reply app/services/fetcher_service.rb]: Fixed.

File: app/services/fetcher_service.rb (hunk 3 above)
Comment (justinlittman) [reply app/services/fetcher_service.rb]: Fixed.

**Pull request merged**